### PR TITLE
Add the path to shiny-server binary and add explanation

### PR DESCRIPTION
**Pull request by vzemlys**
For some reason if the shiny-server is not referenced by the full path , the start-stop-daemon does not start it.

### CODE REVIEW

**Comment by jimklo** [comment]
I know that this is old, but thought I'd still comment, as the suggestion is a misunderstanding of how the ``which`` command works. @antoine-lizee actually that wouldn't be any better than just inserting ``shiny-server``.  ``which`` just returns the absolute path of the first item found on the ``PATH``, as it's possible one might have several versions installed in different locations.  It would still require the the path to the right ``shiny-server`` to be on the ``PATH`` before all others.  If it's not on the ``PATH``, ``which`` won't find it.

TLDR;
Just use an absolute path as the pull request suggests.

**Comment by antoine-lizee** [comment]
Hi @jimklo,

Thank you for trying to teach me that, but I know what which does :-). The only misunderstanding might be about the problem at stake. I had only one version of shiny server installed, but for whatever reason (it's old, as you say it...) using `DAEMON=shiny-server` instead of some kind of `DAEMON=/path/to/shiny-server` didn't work on my machine. It's not about having several versions and all, just about the script failing if you just give the command name, and this is solved with `which`. I might miss something though!

**Comment by jimklo** [comment]
I know your problem has nothing to do with multiple versions, and it could be just the way your system is behaving. According to the man page for 'which', it just searched the PATH environment variable and returns the absolute path of the first one it find. What I'm saying is something is odd about your system if 'which' finds the program, but the shell you're executing 'which' cannot.  They should be using the same PATH environment variable to resolve. Hence why I question using 'which' - I'm not denying it works for you, but I'd like to know why it works, because the way you describe the problem - 'which' should not work either, unless your sub-shell's environment is different from the main shell's. 

If you can do:

'''PATH=  which shiny-server'''

And this returns a path - something is really odd.

**Comment by antoine-lizee** [comment]
Yeah, the problem comes from the fact, as specified by the OP, that if the name of the daemon only is provided, the script will not execute properly. It is because of the following line: 

    [ -x "$DAEMON" ] || exit 0
